extensions/flyway/deployment/src/main/java/io/quarkus/flyway/FlywayProcessor.java

```diff
@@ -223,7 +226,7 @@ ServiceStartBuildItem createBeansAndStartActions(FlywayRecorder recorder,
                     }
                 }
             }
-            return applicationMigrationResources;
+            return new ArrayList<>(applicationMigrationResources);
```

Comment:
Change the return type of this method to `Collection<String>` and you don't need to wrap the `LinkedHashSet` into another collection

Reply:
Sure, but then I need to change all the rest as well. I don't think it's worth it TBH

Reply:
Well, if you look what calls it you'll see it's already expecting `Collection<String>` so it's not a huge change

Reply:
I'll give it another look, but I thought it was List everywhere.

Reply:
Fixed